### ISSUE TITLE
Maybe Clock Drift on Windows Server

### ISSUE THREAD

**Issue by TokenairdropOrg** [prysmaticlabs/prysm#14025]
### Describe the bug

The number of Peers should maintain a dynamic balance,but after running for a period of time, the number of Peers begins to decline unilaterally.

Geth version: 11403
CL client & version: prysm@v5.0.3
OS & Version: Windows server2012

### Has this worked before in a previous version?

```text
This problem did not occur in the past few versions of Prysm software and could run stably for several weeks. However, similar problems occurred in the very early Prysm software (ETH2Beacon had just been launched).

I think the unilateral decline of Peers should be caused by the software not handling the clock drift problem well. This event occurred after Prysm was updated to version v5.0.3（Geth11403), and recurred 4 times. My clock server is time.windows.com (system default)
```

### 🔬 Minimal Reproduction

run geth 11403 + prysm 503 on windows server 2012
wait for some days,maybe 1~5 days.

### Platform(s)

Windows (ARM)

### What version of Prysm are you running? (Which release)

v5.0.3

### Anything else relevant (validator index / public key)?

![image](https://github.com/prysmaticlabs/prysm/assets/79444537/48baa86b-6f48-4401-857d-597d9b87d74a)

**Comment by james-prysm** [comment]
https://github.com/prysmaticlabs/prysm/issues/13431#issuecomment-1891167396 have you tried this?

**Comment by TokenairdropOrg** [comment]
> [#13431 (comment)](https://github.com/prysmaticlabs/prysm/issues/13431#issuecomment-1891167396) have you tried this?

No.I have try to change ntp server to ntp.org.Next time I try to change time.google.com (Maybe a few days later).

**Comment by TokenairdropOrg** [comment]
> [#13431 (comment)](https://github.com/prysmaticlabs/prysm/issues/13431#issuecomment-1891167396) have you tried this?

I try to change NTP server to ntp.org , but the problem still.. Now I try to change NTP server to time.google.com
![image](https://github.com/prysmaticlabs/prysm/assets/79444537/d7e1f544-4662-4b5c-9da1-0b6b801bcd50)
